#! /bin/sh /usr/share/dpatch/dpatch-run
## 2003_possible_crash_draganddrop.patch.dpatch by  <robin@debian>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: 2003_possible_crash_draganddrop.patch

@DPATCH@

--- linden/indra/newview/lltooldraganddrop.cpp	2007-07-11 15:19:48.000000000 +0200
+++ linden/indra/newview/lltooldraganddrop.cpp	2007-08-03 21:09:45.109375000 +0200
@@ -962,6 +962,12 @@
 		{
 			LLInventoryObject* cargo = locateInventory(item, cat);
 
+			if (!cargo) 
+			{
+				handled = FALSE;
+				break;
+			}
+
 			EAcceptance item_acceptance = ACCEPT_NO;
 			handled = handled && root_view->handleDragAndDrop(x, y, mask, FALSE,
 												mCargoTypes[mCurItemIndex],
